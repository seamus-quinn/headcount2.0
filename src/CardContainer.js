import React from 'react';
import Card from './Card';
import './styles/CardContainer.css';
import PropTypes from 'prop-types';

const CardContainer = ({
  schoolData,
  schoolNames,
  updateComparedSchools,
  comparedSchools
}) => {
  const cards = schoolNames.map((schoolName, index) => {
    if (comparedSchools.includes(schoolName)) {
      const cardData = schoolData[schoolName];
      return <Card
        cardData={cardData}
        key={index}
        updateComparedSchools={updateComparedSchools}
        className="card border"
      />;
    } else {
      const cardData = schoolData[schoolName];
      return <Card
        cardData={cardData}
        key={index}
        updateComparedSchools={updateComparedSchools}
        className="card"
      />;
    }
  });
  return cards;
};

CardContainer.propTypes = {
  schoolData: PropTypes.object,
  schoolNames: PropTypes.array,
  updateComparedSchools: PropTypes.func,
  comparedSchools: PropTypes.array
};

export default CardContainer;