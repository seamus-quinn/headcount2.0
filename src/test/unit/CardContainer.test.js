import React from 'react';
import CardContainer from '../../CardContainer';
import { shallow } from 'enzyme';

describe('CardContainer', () => {
  let cardContainer;
  let mockSchoolNames = ['COLORADO', 'ACADEMY 20'];
  let mockComparedSchools = ['COLORADO', 'ACADEMY 20'];
  let mockSchoolData = {
    'COLORADO':
      {
        location: 'COLORADO',
        stats: { 2003: .021 }
      }
  };
  let mockSchoolRepository;
  let mockUpdateComparedSchools;

  beforeEach(() => {
    cardContainer = shallow(<CardContainer
      comparedSchools={mockComparedSchools}
      schoolData={mockSchoolData}
      schoolRepository={mockSchoolRepository}
      updateComparedSchools={mockUpdateComparedSchools}
      schoolNames={mockSchoolNames}
    />);
  });

  it('matches snapshot', () => {

    expect(cardContainer).toMatchSnapshot();
  });
});