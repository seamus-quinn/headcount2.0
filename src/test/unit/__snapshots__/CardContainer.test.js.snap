// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    comparedSchools={
        Array [
            "COLORADO",
            "ACADEMY 20",
          ]
    }
    schoolData={
        Object {
            "COLORADO": Object {
              "location": "COLORADO",
              "stats": Object {
                "2003": 0.021,
              },
            },
          }
    }
    schoolNames={
        Array [
            "COLORADO",
            "ACADEMY 20",
          ]
    }
    schoolRepository={undefined}
    updateComparedSchools={undefined}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Card
      cardData={
            Object {
                  "location": "COLORADO",
                  "stats": Object {
                    "2003": 0.021,
                  },
                }
      }
      className="card border"
      updateComparedSchools={undefined}
/>,
    <Card
      cardData={undefined}
      className="card border"
      updateComparedSchools={undefined}
/>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Card
        cardData={
                Object {
                        "location": "COLORADO",
                        "stats": Object {
                          "2003": 0.021,
                        },
                      }
        }
        className="card border"
        updateComparedSchools={undefined}
/>,
      <Card
        cardData={undefined}
        className="card border"
        updateComparedSchools={undefined}
/>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
